src/Features/Core/Portable/Completion/Providers/AbstractObjectCreationCompletionProvider.cs

```diff
@@ -79,48 +80,79 @@ protected override Task<ImmutableArray<ISymbol>> GetSymbolsWorker(SyntaxContext
 
             if (type.SpecialType == SpecialType.System_Void)
             {
-                return SpecializedTasks.EmptyImmutableArray<ISymbol>();
+                return ImmutableArray<(ISymbol, CompletionItemRules)>.Empty;
             }
 
             if (type.ContainsAnonymousType())
             {
-                return SpecializedTasks.EmptyImmutableArray<ISymbol>();
+                return ImmutableArray<(ISymbol, CompletionItemRules)>.Empty;
             }
 
             if (!type.CanBeReferencedByName)
             {
-                return SpecializedTasks.EmptyImmutableArray<ISymbol>();
+                return ImmutableArray<(ISymbol, CompletionItemRules)>.Empty;
+            }
+
+            if (type.IsInterfaceType())
+            {
+                var typeArity = type.GetArity();
+
+                var implementations = await SymbolFinder.FindImplementationsAsync(
+                    type, context.Workspace.CurrentSolution, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                var matchingImplementations = implementations.Where(
+                    impl => !impl.IsAbstract && impl.GetArity() == typeArity).Cast<INamedTypeSymbol>();
+
+                var userAssembly = context.SemanticModel.Compilation.Assembly;
+
+                matchingImplementations = matchingImplementations.Where(
+                    impl => impl.InstanceConstructors.Any(c => c.IsAccessibleWithin(userAssembly)));
+
+                if (typeArity > 0)
+                {
+                    var typeArguments = type.GetTypeArguments();
+
+                    var constructedImplementations = ArrayBuilder<(ISymbol, CompletionItemRules)>.GetInstance();
+
+                    foreach (var impl in matchingImplementations)
+                    {
+                        constructedImplementations.Add((impl.Construct(typeArguments.ToArray()), CompletionItemRules.Default));
```

Comment:
Yes, I'm just try to lay the groundwork for what's coming next and because I'm unfamiliar with the codebase, I'm taking baby steps. :)